scanner/src/main/kotlin/experimental/ExperimentalScanner.kt

```diff
@@ -392,7 +392,7 @@ class ExperimentalScanner(
                 }
             }
 
-            result
+            result.mapValuesTo(mutableMapOf()) { it.value.toList() }
```

Comment:
This is quite ugly, but I currently cannot think of a better solution either.

Reply:
Me neither, I tried a few things but nothing worked.

Reply:
To get rid of this, would it be possible to generate the result map in a more functional way, e.g. something in the lines of:

```
provenances.associateWith { provenance ->
    storageReaders.filterIsInstance(...)
      .flatMap { reader ->
         reader.read(...)
      }
}
```

Reply:
The problem is that there are two different types of storage readers, combining both in a single loop would be very difficult.

Reply:
Thanks, no objections from my side.